apps/frontend/src/libs/helpers/progress-bar-steps/progress-bar-steps.ts

```diff
@@ -0,0 +1,27 @@
+import { type Step } from "~/libs/components/components.js";
+
+const FIRST_STEP_INDEX = 0;
+const NEXT_STEP_OFFSET = 1;
+
+const createSteps = (questionsLength: number): Step[] => {
+	return Array.from({ length: questionsLength }, (_, index) => ({
+		status: index === FIRST_STEP_INDEX ? "current" : "upcoming",
+		step: index + NEXT_STEP_OFFSET,
+	}));
+};
+
+const advanceStep = (steps: Step[]): Step[] => {
+	const currentStepIndex = steps.findIndex((step) => step.status === "current");
+
+	return steps.map((step, index) => {
+		if (index === currentStepIndex) {
+			return { ...step, status: "completed" };
+		} else if (index === currentStepIndex + NEXT_STEP_OFFSET) {
+			return { ...step, status: "current" };
+		}
+
+		return step;
+	});
+};
```

Comment:
I don't think we need those, this makes the logic more complex. What about keeping only `step`, without any status? Step will change one by one

Reply:
In Figma, we have different statuses for the Status bar. (in the quiz page and onboarding page).
<img width="422" alt="Screenshot 2024-08-27 at 21 41 31" src="https://github.com/user-attachments/assets/10ade002-c434-4d1a-b984-e5588d4125d8">
<img width="415" alt="Screenshot 2024-08-27 at 21 41 21" src="https://github.com/user-attachments/assets/84001197-97db-471b-82fa-af9aec734142">

So I made this component common and used this complex logic

Reply:
We can handle this by knowing current question index, can't we? Number of steps = number of questions. Completed steps = index < current index. Current step = index === current index